string first_goal
int8 length
---
string[] goal_list